Implement len() operator for embed

zykrahgaming opened the pull request:
### Summary
<!-- Small summary of the merge request -->
Request to implement a `len()` operator for `Embed` objects, reasoning being that discord limits embeds to 6000 total characters (counting title, description, all field names and values, the author and footer) on top of the pre-existing limits for fields, descriptions, etc... This operator makes it simple check if any embed object exceeds the character limit before sending the actual message. It would then be up to the user to decide what to do to (ideally) keep the embed under the character limit. The operator also exists in other libraries such as `discord.py`.

Thanks to @norinorin for helping me out with the code, as I initially had some difficulty involving the `Optional[]` nature of the attributes.

### Checklist
<!-- Make sure to tick all the following boxes by putting an `x` in between (like this `[x]`) -->
- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### Related issues
<!--
To mention an issue use `#issue-id` and to mention a merge request use `!merge-request-id`
To close/fix an issue use `Close #issue-id` or `Fix #issue-id` (depending on the merge request)
-->

davfsa commented:
I don't think this is needed at all. This value can be easily calculated outside of the class, as well as it doesnt provide any more information.

Furthermore, `len()` is probably not the best way to implement this, as embeds dont technically have a length.

ben2224 commented:
> Furthermore, `len()` is probably not the best way to implement this, as embeds dont technically have a length.

could just call it Embed.total_length() or something

davfsa reviewed (approved):
Just this little thing and LGTM

zykrahgaming commented on hikari/embeds.py:
```diff
@@ -991,3 +991,18 @@ def __eq__(self, other: typing.Any) -> bool:
                 return True
         return False
+
+    def __len__(self) -> int:
```
Changed

zykrahgaming commented:
Changed function name to `total_length()` as suggested, and added a description for the function. Let me know if anything is wrong with it.

Forgot about the changelog framework, will add now.

zykrahgaming commented:
Renamed branch for consistency. Will open a new pull request and edit as needed.
